tests/test_tokenization.py

```diff
@@ -9,12 +15,303 @@ def _assert_tokenization(tokenizer: TokenizerWrapper):
     assert len(token_ids) > 0
 
 
-def test_hf_tokenize():
-    tokenizer_model_file = "data/tokenizer/hf_gpt2"
-    tokenizer = PreTrainedHFTokenizer(
-        pretrained_model_name_or_path=tokenizer_model_file, max_length=20, truncation=False, padding=False
+@pytest.mark.parametrize(
+    "text,tokenizer_config,expected_length,expected_num_paddding_tokens",
+    [
+        # Test cases 1: Sequence is shorter than max_length, i.e., len(text) < max_length
+        # If padding="max_length", we want a sequence to be padded to the max_length, irrespective of the truncation flag
+        # and only if max_length is specified.
+        # If max_length is not specified, we pad to the max model input length (i.e., 1024 for the gpt2 model).
+        # NOTE: "AAAAAAAA" is a single token for the gpt2 tokenizer, there is "A" sequence longer than that in the vocabulary
+        (
+            "AAAAAAAA" * 6,
+            PreTrainedHFTokenizerConfig(
+                pretrained_model_name_or_path="data/tokenizer/hf_gpt2",
+                truncation=False,
+                padding="max_length",
+                max_length=10,
+                special_tokens={"pad_token": "[PAD]"},
+            ),
+            10,
+            4,
+        ),
```

Comment:
The `PreTrainedHFTokenizerConfig` is used with the same arguments `pretrained_model_name_or_path` and `special tokens` for all test cases. It would perhaps be simpler and more readable if only `truncation`, `padding` and `max_length` were used as input arguments, with `PreTrainedHFTokenizerConfig` being instantiated in the test function below.